### ISSUE TITLE
experimental:spacing-between-declarations-with-comments false positive

### ISSUE THREAD

**Issue by romtsn** [pinterest/ktlint#865]
## Expected Behavior
Expect this code to not cause any issues
```Kotlin
fun function() {
    with(binding) {
        loginButton.setOnClickListener {
//            comment
            showScreen()
        }
    }
}
```

## Observed Behavior
```Kotlin
path/to/file/File.kt:4:1: Declarations and declarations with comments should have an empty space between. (experimental:spacing-between-declarations-with-comments)
```

## Steps to Reproduce
Run lint on the given code snippet, in case of formatting - an empty line will be inserted before the `// comment`

## Your Environment

* Version of ktlint used: 0.38.0
* Name and version (or code for custom task) of integration used (Gradle plugin, Maven plugin, command line, custom Gradle task): CLI
